hs_composite_resource/models.py

```diff
@@ -0,0 +1,180 @@
+import os
+
+from mezzanine.pages.page_processors import processor_for
+
+from hs_core.models import BaseResource, ResourceManager, resource_processor
+
+from hs_file_types.models import GenericLogicalFile
+
+
+class CompositeResource(BaseResource):
+    objects = ResourceManager("CompositeResource")
+
+    class Meta:
+        verbose_name = 'Composite Resource'
+        proxy = True
+
+    @property
+    def can_be_public_or_discoverable(self):
+        # resource level metadata check
+        if not super(CompositeResource, self).can_be_public_or_discoverable:
+            return False
+
+        # filetype level metadata check
+        for lf in self.logical_files:
+            if not lf.metadata.has_all_required_elements():
+                return False
+
+        return True
+
+    def set_default_logical_file(self):
+        """sets an instance of GenericLogicalFile to any resource file objects of this instance
+        of the resource that is not already associated with a logical file. """
+
+        for res_file in self.files.all():
+            if not res_file.has_logical_file:
+                logical_file = GenericLogicalFile.create()
```

Comment:
@pkdash I prefer to use key/value for metadata if possible as well. If you think you can get it done in time, it might be less work and easier to go with key/value metadata here rather than try to reuse existing code as much as possible since existing code is going to be changed to key/value metadata anyway. I'll defer this to you but feel like going with key/value metadata here may be simpler and easier than what you have done here.